tsec-http4s/src/main/scala/tsec/authentication/JWKPublicKeyRSAAuthenticator.scala

```diff
@@ -43,17 +43,22 @@ class KeyRegistry[F[_], A: JCASigTag: RSAKFTag](
    client: Client[F]
 )(implicit E: Effect[F]) {
 
-  implicit val jwksDecoder: EntityDecoder[F, JWKS] = jsonOf[F, JWKS]
-  implicit val modulusDecoder: Decoder[Modulus]    = Decoder.decodeString.map(s => Modulus(BigInt(1, s.base64UrlBytes)))
-  implicit val exponentDecoder: Decoder[Exponent]  = Decoder.decodeString.map(s => Exponent(BigInt(1, s.base64UrlBytes)))
+  implicit val jwksDecoder: EntityDecoder[F, JWKS[A]] = jsonOf[F, JWKS[A]]
+  implicit val modulusDecoder: Decoder[Modulus]       = Decoder.decodeString.map(s => Modulus(BigInt(1, s.base64UrlBytes)))
+  implicit val exponentDecoder: Decoder[Exponent]     = Decoder.decodeString.map(s => Exponent(BigInt(1, s.base64UrlBytes)))
+  implicit val x5cDecoder: Decoder[SigPublicKey[A]]   = Decoder.decodeString.map(s => {
```

Comment:
@jmcardon would this be a better way to get the public key? Does it make sense to additionally compare `n` and `e` from the JWK with the values of the generated cert?